frontend/src/store/index.js

```diff
@@ -395,19 +395,37 @@ const getters = {
       return max(map(seedsMatchingCloudProfileAndRegion, 'volume.minimumSize')) || defaultMinimumSize
     }
   },
-  floatingPoolNamesByCloudProfileNameAndRegion (state, getters) {
-    return (cloudProfileName, region) => {
+  floatingPoolNamesByCloudProfileNameAndRegionAndDomain (state, getters) {
+    return ({ cloudProfileName, region, secretDomain }) => {
       const cloudProfile = getters.cloudProfileByName(cloudProfileName)
       const floatingPools = get(cloudProfile, 'data.providerConfig.constraints.floatingPools')
-      const availableFloatingPools = filter(floatingPools, matchesPropertyOrEmpty('region', region))
+      let availableFloatingPools = filter(floatingPools, matchesPropertyOrEmpty('region', region))
+      availableFloatingPools = filter(availableFloatingPools, matchesPropertyOrEmpty('domain', secretDomain))
+
+      if (find(availableFloatingPools, fp => {
+        return !!fp.region && !fp.nonConstraining
+      })) {
+        availableFloatingPools = filter(availableFloatingPools, { region })
+      }
+      if (find(availableFloatingPools, fp => {
+        return !!fp.domain && !fp.nonConstraining
+      })) {
+        availableFloatingPools = filter(availableFloatingPools, { domain: secretDomain })
```

Comment:
why not call secretDomain just domain?

Reply:
I called it that way to make clear where the domain is coming from. I don't know if there are other places where domains can be specified e.g. in the cloud profile.

Reply:
okay let's leave it as is